relay-cardinality/src/limiter.rs

```diff
@@ -208,7 +214,20 @@ impl<'a> Reporter<'a> for DefaultReporter<'a> {
     fn reject(&mut self, limit: &'a CardinalityLimit, entry_id: EntryId) {
         self.exceeded_limits.insert(limit);
         if !limit.passive {
-            self.entries.insert(entry_id.0);
+            match self.entries.entry(entry_id.0) {
+                hashbrown::hash_map::Entry::Occupied(entry) => {
+                    let existing_limit = entry.get();
+                    // Scopes are ordered by reverse specificity (org is the smallest), so we use `Reverse` here
+                    if (Reverse(limit.scope), limit.limit)
+                        < (Reverse(existing_limit.scope), existing_limit.limit)
+                    {
+                        entry.replace_entry(limit);
+                    }
+                }
+                hashbrown::hash_map::Entry::Vacant(entry) => {
+                    entry.insert(limit);
+                }
```

Comment:
If we implement `Ord` on the limit we could do this:

```rs
self.entries.entry(..).and_modify(|entry| *entry = (*entry).min(new)).or_insert(new);
```

Without implementing `Ord` on the limit (which may be a bit weird), we can still use the combinators:

```rs
self.entries.entry(..).and_modify(|entry| if a < b { *entry = new }).or_insert(new);
```

I prefer the combinators but don't have a strong opinion.